src/app/services/source.ts

```diff
@@ -1,53 +1,110 @@
-const pos_precision = 3;
 
-export class SourceTeV {
+class SourceBase {
+
   public data;
   constructor(data) {
     this.data = data;
   }
 
-  // ra_str() {
-  //     return this.ra.toFixed(pos_precision) + ' deg';
-  //   //   return this._pos_with_err_str('RA');
-  // }
-  // dec_str() {
-  //     return this.dec.toFixed(pos_precision) + ' deg';
-  //   //   return this._pos_with_err_str('DEC');
-  // }
-  // glon_str() { return this.glon.toFixed(pos_precision) + ' deg'; }
-  // glat_str() { return this.glat.toFixed(pos_precision) + ' deg'; }
+  formatf(val) {
+    // Value has fixed number of digits
+    if(val != null)
+      return val.toFixed(3);
+    else
+      return val;
+  }
+
+  formatp(val) {
+    // Value to a certain precision
+    if(val != null)
+      return val.toPrecision(3);
+    else
+      return val;
+  }
+
+
+  format_error(val, val_err) {
+    return `${this.formatp(val)} +- ${this.formatp(val_err)}`;
+  }
+
+}
+
+export class SourceTeV extends SourceBase {
 
   class_str() {
     return this.data.classes.split(",").join(", ");
   }
+
+  gamma_names_str() {
+    return this.data.gamma_names.split(",").join(", ");
+  }
+  fermi_names_str() {
+    return this.data.fermi_names.split(",").join(", ");
+  }
   other_names_str() {
     return this.data.other_names.split(",").join(", ");
   }
-  gamma_names_str() {
-    return this.data.gamma_names.split(",").join(", ");
+
+  get_spec_parameters() {
+    var spec = this.data.spec_type;
+    var markup;
+
+    if(spec == 'pl') {
+      markup = `
+      <ul>
+        <li>Spectrum type: pl</li>
+        <li>norm: ${this.format_error(this.data.spec_pl_norm, this.data.spec_pl_norm_err)} cm-2 s-1 TeV-1 (statistical)</li>
```

Comment:
This is what you have now:
<img width="407" alt="screen shot 2017-07-13 at 13 48 18" src="https://user-images.githubusercontent.com/852409/28164940-f5987110-67d1-11e7-9385-7d9914b27dbc.png">

You should only have one line for each parameter, which is (hopefully) how it's already in the Python / Gammapy printout.
"reference" is the reference energy in TeV.